### DISCUSSION TITLE
Inserting object data into local config context?

### DISCUSSION THREAD

**Discussion by llamafilm** [netbox-community/netbox#14875] [Help Wanted!]
I saw this interesting [presentation](https://nog.fi/event/8/contributions/76/attachments/59/119/NTxG%20Presentation.pdf) from an engineer at Nokia, describing how they use Netbox config contexts to generate Telegraf configs.  Unfortunately I could not understand much of the [audio](https://www.youtube.com/watch?v=9iiU8PDdxdo&feature=youtu.be).

I'm trying to figure out how this would work.  If I want to monitor device health using Telegraf SNMP input plugin, the config needs to know the IP address of each device.  So I could put most of the SNMP config in the Device Model config context, and then I would need a way to insert the device's Primary IP address using local config context.  Is this possible?

**Comment by llamafilm** [comment]
I did a little testing and found that `get_config_context()` is fairly slow when iterating though a large number of devices.  Instead, I think it will be more efficient to store text files on disk for each Telegraf template, and use Netbox script to replace strings and build the full config that way.

**Comment by candlerb** [comment]
I haven't watched the video, but for the device's IP address I'd suggest you use its primary_ip address attribute.

(If you are configuring a device from Netbox data, the information you need doesn't *all* have to come from config_context data. This is mainly intended for those settings which are often shared between all devices in a site, rack, tenant etc. Things which are set individually on a device could be standard Netbox attributes or custom fields)
